enricher/enricher.go

```diff
@@ -0,0 +1,117 @@
+package enricher
+
+import (
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// EnrichRecord modifies existing record.
+func EnrichRecord(r map[interface{}]interface{}, t time.Time) map[interface{}]interface{} {
+	return defaultEnricher.enrichRecord(r, t)
+}
+
+type enricher struct {
+	enable          bool
+	canvaAWSAccount string
+	canvaAppName    string
+	logGroup        string
+	ecsTaskFamily   string
+	ecsTaskRevision int
+}
+
+var defaultEnricher *enricher
+
+// Init will initialise defaultEnricher instance.
+func Init(enable bool) {
+	if !enable {
+		defaultEnricher = new(enricher)
+		return
+	}
+
+	ecsTaskDefinition := os.Getenv("ECS_TASK_DEFINITION")
+	re := regexp.MustCompile(`^(?P<ecs_task_family>[^ ]*):(?P<ecs_task_revision>[\d]+)$`)
```

Comment:
IIRC golang has regexp performance problems. It might be easier to split this with `:` instead of using `regexp` 😄 

https://www.reddit.com/r/golang/comments/84o986/why_is_gos_regex_so_slow/

Reply:
I use it only once in initialising the plugin, not after that. Performance is not important really :-P